keep replicas when autoscaler set external

Jooho opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:
1. If this is your first time, read our contributor guidelines https://www.kubeflow.org/docs/about/contributing/ and developer guide https://github.com/kserve/kserve/blob/master/docs/DEVELOPER_GUIDE.md
2. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
3. Follow the instructions for writing a release note: https://git.k8s.io/community/contributors/guide/release-notes.md
4. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

**What this PR does / why we need it**:
When autoscaler set external, we should not set nil for replica in deployment. 
If we set nil, it will be set 1 always when isvc is updated.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Type of changes**
Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Feature/Issue validation/testing**:

Please describe the tests that you ran to verify your changes and relevant result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

- Logs

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Checklist**:

- [ ] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

**Release note**:
<!--  Write your release note:
1. Enter your extended release note in the below block. If the PR requires additional action from users switching to the new release, include the string "action required".
2. If no release note is required, just write "NONE".
-->
```release-note

```

**Re-running failed tests**

- `/rerun-all` - rerun all failed workflows.
- `/rerun-workflow <workflow name>` - rerun a specific failed workflow. Only one workflow name can be specified. Multiple /rerun-workflow commands are allowed per comment.

Jooho committed:
keep replicas when autoscaler set external

Signed-off-by: jooho lee <jlee@redhat.com>

Jooho committed:
fix golint

Signed-off-by: jooho lee <jlee@redhat.com>

spolti reviewed (commented):
can we have a simple unit test for this?

Jooho commented:
/rerun-all

spolti commented:
/rerun-workflow E2E Tests / test-qpext

Jooho commented:
/rerun-all

Jooho committed:
Update replicas when autoscaler is external and minReplica set.
Do not ignore field "replicas" to diff when autoscaler is external.
update go.mod go.sum

Signed-off-by: jooho lee <jlee@redhat.com>

Jooho commented:
@spolti added unit tests

spolti commented on go.mod:
```diff
@@ -11,6 +11,7 @@ require (
 	github.com/go-logr/logr v1.4.2
 	github.com/go-logr/zapr v1.3.0
 	github.com/gofrs/uuid/v5 v5.3.0
+	github.com/golang/mock v1.4.4
```
looks like `go mod tidy` is needed.

Jooho replied:
updated.

spolti reviewed (commented):
/lgtm

code looks good, however, the `verify-go-mod` GHA is complaining.

Jooho committed:
git mod tidy done

Signed-off-by: jooho lee <jlee@redhat.com>

sivanantha321 committed:
Merge branch 'master' into external_autoscale

sivanantha321 commented:
/lgtm
/approve
